experimental/http_logger/http_logger.go

```diff
@@ -43,6 +43,19 @@ func NewHTTPLogger(pluginID string, proxied http.RoundTripper) *HTTPLogger {
 	}
 }
 
+// WithPath sets the path to store HAR file.
```

Comment:
Apologies if I mentioned this before, but did you consider something like:

```golang

type Opts struct {
	Path      string
	EnabledFn func() bool
}

// NewHTTPLogger creates a new HTTPLogger.
func NewHTTPLogger(pluginID string, proxied http.RoundTripper, opts ...Opts) *HTTPLogger {
	var loggerOpts Opts
	switch len(opts) {
	case 0:
		loggerOpts = Opts{
			Path:      defaultPath(pluginID),
			EnabledFn: defaultEnabledCheck,
		}
	case 1:
		loggerOpts = opts[0]
	default:
		panic("only an empty or one Opts is valid as argument")
	}
	
	// TODO validate loggerOpts

	s := storage.NewHARStorage(loggerOpts.Path)
	f := fixture.NewFixture(s)

	return &HTTPLogger{
		pluginID: pluginID,
		proxied:  proxied,
		fixture:  f,
		enabled:  loggerOpts.EnabledFn,
	}
}
```

If I understand the use correctly, this should make it easier to construct from the client without having to create a logger, and then edit directly. WDYT?

Reply:
nice, i'll change to that approach. thanks!

Reply:
added with [d4f0a37](https://github.com/grafana/grafana-plugin-sdk-go/pull/485/commits/d4f0a37ce5a4f90aafa2469048c2fbf7e2bbff29)

Reply:
Looks great! I assume you can remove the `WithPath` and `WithEnabledCheck` in that case?

Reply:
🤦 hah, missed those. removed in [9429436](https://github.com/grafana/grafana-plugin-sdk-go/pull/485/commits/9429436d499a1b331af6d26ae59222fb1d320dfb)